lib/id3c/cli/command/geocode.py

```diff
@@ -411,6 +412,34 @@ def parse_first_smartystreets_result(result: list) -> dict:
     return response
 
 
+def extract_address(address: dict) -> dict:
+    """
+    Given an *address*, converts it to text and returns a result from
+    the SmartyStreets US Extract API containing information about an address
+    connected to the text.
+
+    Note that this API is not consistent with the US Street API, and the lookup
+    and responses must be handled differently.
+    """
+    LOG.debug("Previous lookup failed. Looking up address as free text.")
+
+    client = smartystreets_client_builder().build_us_extract_api_client()
+    address_text = ', '.join([str(val) for val in list(address.values()) if val])
+
+    lookup = ExtractLookup()
+    lookup.text = address_text
+
+    result = client.send(lookup)
+    addresses = result.addresses
+
+    for add in addresses:
+        if len(add.candidates) > 0:
+            result = add.candidates
+            return parse_first_smartystreets_result(result)
+
+    return {}
```

Comment:
The other geocoding function returns `None` when it doesn't succeed instead of an empty dict. Would be nice to be consistent.